### PR TITLE
feat: [CDS-71711]: add github, nexus3 & artifactory digest  

### PR DESCRIPTION
**Pull request by namanharness**
###  Services
https://github.com/harness/harness-core-ui/assets/110456765/f8717d8e-5990-4b26-bd11-4b7cf63ccd95

### Artifact Source template & Services in Pipeline Studio

https://github.com/harness/harness-core-ui/assets/110456765/c4d7f8fe-88ca-43dd-a457-972199931d51

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Nexus3Artifact/Nexus3Artifact.tsx [hunk 1]
```diff
@@ -1127,6 +1139,16 @@ export function Nexus3Artifact({
                 isImagePath={false}
                 tooltipId="nexus3-tag"
               />
+              {CD_NG_DOCKER_ARTIFACT_DIGEST && formik.values?.repositoryFormat === RepositoryFormatTypes.Docker && (
```
Comment (namanharness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Nexus3Artifact/Nexus3Artifact.tsx:1142]: digest added only for repo format - docker

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Artifactory/Artifactory.tsx [hunk 2]
```diff
@@ -371,11 +372,13 @@ function Artifactory({
       getMultiTypeFromValue(artifactFormDataValues.artifactPath) === MultiTypeInputType.FIXED &&
       (artifactFormDataValues.artifactPath as string)?.length
     ) {
-      artifactFormDataValues.artifactPath = {
-        label: artifactFormDataValues?.artifactPath,
-        value: artifactFormDataValues?.artifactPath
-      } as SelectOption
+      artifactFormDataValues.artifactPath = getLabelValueObject(artifactFormDataValues.artifactPath)
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Artifactory/Artifactory.tsx:375]: please confirm if "artifactFormDataValues" can be undefined or not

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Artifactory/Artifactory.tsx [hunk 3]
```diff
@@ -815,6 +822,16 @@ function Artifactory({
                     )}
                   </div>
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Artifactory/Artifactory.tsx:823]: Please test the scenario where a user might chose docker repo format - then configure digest and then change to generic repo format --> in this case the yaml should not contain digest field

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactUtils.ts [hunk 4]
```diff
@@ -472,13 +501,24 @@ export const getArtifactFormData = (
   if (selectedArtifact === ENABLED_ARTIFACT_TYPES.GoogleArtifactRegistry) {
     values = getGarDigestValues(values as GoogleArtifactRegistryInitialValuesType)
   }
+  if (selectedArtifact === ENABLED_ARTIFACT_TYPES.GithubPackageRegistry) {
+    values = getGarDigestValues(values as GithubPackageRegistryInitialValuesType) // also work for Github Package Registry
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactUtils.ts:505]: Also works for*

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Artifactory/Artifactory.tsx (hunk 3 above)
Comment (namanharness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Artifactory/Artifactory.tsx:823]: i have added code to reset digest field when repo format is toggled and
in the code at submit level when when repo format is generic we delete the digest key , does this work ?

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Artifactory/Artifactory.tsx (hunk 2 above)
Comment (namanharness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Artifactory/Artifactory.tsx:375]: No it should not be , it is returned from 'getArtifactFormData' function

**Review by samarth-sharma31 — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

**Pull request merged**